src/include/distributed/version_compat.h

```diff
@@ -184,6 +189,23 @@ FileCompatFromFileStart(File fileDesc)
 }
 
 
+/*
+ * postgres 11 equivalent for a function with the same name in postgres 12+.
+ */
+static inline bool
+table_scan_getnextslot(HeapScanDesc scan, ScanDirection dir, TupleTableSlot *slot)
+{
+	HeapTuple tuple = heap_getnext(scan, ForwardScanDirection);
+	if (tuple == NULL)
+	{
+		return false;
+	}
+
+	ExecStoreTuple(tuple, slot, InvalidBuffer, false);
+	return true;
+}
+
+
```

Comment:
I like this a lot! Was thinking ifdefs in the loop where this is called, but this is way cleaner! Thanks